### PR TITLE
IR-1830 - Add project id to location and instance table

### PR DESCRIPTION
**Pull request by hurairahmateen**
## Summary
- Add project id to location and instance table.
- Created migrations to add projectId column.
- Updated service calls to include projectId

### CODE REVIEW

File: packages/server-core/src/networking/instance/instance.resolvers.ts [hunk 1]
```diff
@@ -49,6 +56,28 @@ export const instanceDataResolver = resolve<InstanceType, HookContext>({
   id: async () => {
     return uuidv4() as InstanceID
   },
+  projectId: async (value, instance, context) => {
+    try {
+      // Populate projectId from locationId
+      if (instance.locationId) {
+        const locationData = await context.app.service(locationPath).get(instance.locationId)
+        if (locationData) {
```
Comment (hanzlamateen) [inline packages/server-core/src/networking/instance/instance.resolvers.ts:63]: If there is no locationData then it should throw an error.

File: packages/server-core/src/networking/instance/instance.resolvers.ts [hunk 2]
```diff
@@ -49,6 +56,28 @@ export const instanceDataResolver = resolve<InstanceType, HookContext>({
...
+      if (instance.channelId) {
+        const channelData = await context.app.service(channelPath).get(instance.channelId)
+        if (channelData.instanceId) {
+          const channelInstance = await context.app.service(instancePath).get(channelData.instanceId)
+          return channelInstance.projectId
+        }
+        return ''
+      }
+    } catch (error) {
+      throw new BadRequest('Error populating projectId into instance')
```
Comment (hanzlamateen) [inline packages/server-core/src/networking/instance/instance.resolvers.ts:79]: Do a log error of `instance` object here for later debugging.

File: packages/server-core/src/social/location/location.resolvers.ts [hunk 3]
```diff
@@ -99,6 +101,14 @@ export const locationDataResolver = resolve<LocationType, HookContext>({
       updatedAt: await getDateTimeSql()
     }
   },
+  projectId: async (value, location, context) => {
+    try {
+      const scene = await context.app.service(assetPath).get(location.sceneId)
+      return scene.projectId
+    } catch (error) {
+      throw new BadRequest('Failed to retrieve project ID from scene ID')
```
Comment (hanzlamateen) [inline packages/server-core/src/social/location/location.resolvers.ts]: Add `location.sceneId` to this message

**Review by hanzlamateen — CHANGES_REQUESTED** [review]

**Review by hanzlamateen — APPROVED** [review]
Looks good to me. @HexaField can you give it a look as well.

**Comment by hurairahmateen** [comment]
Done through MT now.

https://github.com/theinfinitereality/eepro-multitenancy/pull/212

**Review by hanzlamateen — APPROVED** [review]

**Pull request merged**